lib/cext/preprocess.rb

```diff
@@ -783,6 +783,103 @@ def tuple_new_patch(ctx, slf)
         replacement: 'if (rb_tr_obj_equal(Qtrue, obj))'
       }
     ]
+  },
+  'database.c' => {
+    gem: 'sqlite3',
+    patches: [
+      { # tracefunc
+        match: 'VALUE self = (VALUE)data;',
+        replacement: 'VALUE self = rb_tr_managed_from_handle(data);'
+      },
+      { # trace
+        match: 'sqlite3_trace(ctx->db, NIL_P(block) ? NULL : tracefunc, (void *)self);',
```

Comment:
Consider a regex replacement for `(void *)self`, this one appears to repeat a few times.

Reply:
I really prefer matching the exact string. I think the regexp matches we do are major headache waiting to happen.

Reply:
yeah we should think about replacing the regex with diff. It would be more exact and it also little bit more understandable in the source code what is being changed.

Reply:
A diff would fail very quickly to apply with any change around though.
I'd rather focus on not needing these patches.